content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Don't Just Track Your Experiments, Version Them
+date: 2021-11-17
+description: >
+  Versioning ML experiments in DVC helps you log, organize, and reproduce
+  experiments easier than ever.
+descriptionLong: >
+  With DVC experiments, you can take ML experiment tracking to the next level,
+  making it easy and powerful to save, compare, and reproduce experiments at
+  scale.
+picture:
+pictureComment:
+author: dave_berenbaum
+commentsUrl:
+tags:
+  - MLOps
+  - DVC
+  - Experiments
+  - Experiment Tracking
+  - Experiment Versioning
+---
+
+Traditional software version control is not enough for tracking machine learning
+experiments. Over the last few years, experiment tracking has come a long way,
+but there are parallels to problems from earlier generations of version control.
+That's why "experiment versioning" in DVC tracks experiments by building on
+modern version control practices.
+
+# Experiment Tracking Needs
+
+When tracking ML experiments, you need to:
+
+- Log experiment information.
+- Organize and compare experiments.
+- Reproduce experiments.
+
+Tracking data, comparing versions, and enabling reproducibility are already
+addressed by version control. Unfortunately, traditional version control tools
+like Git don't work well for ML experimentation because:
+
+- Data and models aren't easily tracked with Git.
+- There are too many experiments to track with any branching workflow.
```

Comment:
Probably a silly question, but we still use branches for tracking experiments with DVC, don't we?  If so, the problem of "too many branches" is still there

Reply:
`dvc exp` uses Git references in the form `refs/exps/123ABCD/exp-4484a` for experiments. These are outside of the branch structure, but can be moved to Git via `dvc exp branch`. The user doesn't think of a way to create branches for experiments with DVC, selects them for branches afterwards.

Reply:
We sometimes call them lightweight branches or hidden branches. The point is that the user does not need to manually generate commits nor branches, nor do they see those experiments when using typical git commands like `git branch`, and there are commands like `dvc exp remove` to clean them up.